Code/Tools/ProjectManager/Source/AddRemoteProjectDialog.cpp

```diff
@@ -159,17 +166,58 @@ namespace O3DE::ProjectManager
             }
         );
 
-        // Simulate repo being entered and UI enabling
         connect(
             m_repoPath->lineEdit(), &QLineEdit::textEdited,
-            [this](const QString& text)
+            [this]([[maybe_unused]] const QString& text)
             {
-                SetDialogReady(!text.isEmpty());
+                // wait for a second before attempting to validate so we're less likely to do it per keypress
+                m_inputTimer->start(1000);
+                
             });
 
         SetDialogReady(false);
     }
 
+    void AddRemoteProjectDialog::ValidateURI()
+    {
+        // validate URI, if it's a valid repository, get the project info and set the dialog as ready
+        bool validRepository = false;
+
+        auto repoProjectsResult = PythonBindingsInterface::Get()->GetProjectsForRepo(m_repoPath->lineEdit()->text());
+        if (repoProjectsResult.IsSuccess())
+        {
+            auto repoProjects = repoProjectsResult.GetValue();
+            if (!repoProjects.isEmpty())
+            {
+                // only get the first one for now
+                const ProjectInfo& project = repoProjects.at(0);
+                SetCurrentProject(project);
+
+                validRepository = true;
+            }
+        }
+        SetDialogReady(validRepository);
+    }
+
+    void AddRemoteProjectDialog::DownloadObject()
+    {
+        // Add Repo:
+        const QString repoUri = m_repoPath->lineEdit()->text();
+        auto addGemRepoResult = PythonBindingsInterface::Get()->AddGemRepo(repoUri);
+        if (addGemRepoResult.IsSuccess())
+        {
+            // Send download to project screen to initiate download
+            emit StartObjectDownload(m_currentProject.m_projectName);
+            emit QDialog::accept();
+        }
+        else
+        {
+            QString failureMessage = tr("Failed to add gem repo: %1.").arg(repoUri);
```

Comment:
nit - I am not sure this translation will ever work because it includes a dynamic URL